### PR TITLE
Updating libsodium to use the right version (libsodium23 over libsodium18)

### LINKED ISSUE #261: Wrong version of libsodium installed (libsodium18 instead of libsodium23)
It appears the sodium library version is outdated.

This causes issues for applications that require newer functions (like built in Argon2 support)

Currently used version: `1.0.8` / `libsodium18`
Latest release: `1.0.16` / `libsodium23`

Example: symfony/symfony#26179

### PR DESCRIPTION
**Pull request by iSDP**
fixes / closes #261

### CODE REVIEW

File: support/build/php [hunk 1]
```diff
@@ -109,6 +109,7 @@ if [[ $dep_version == 7.* ]]; then
     elif [[ $STACK != "cedar-14" ]]; then
         configureopts="\
```
Comment (stof) [inline support/build/php]: @dzuelke isn't this code loosing the `--enable-opcache-file` which was first added  for PHP 7+ above ?

Comment (iSDP) [reply support/build/php]: This was already there 🤔

Comment (stof) [reply support/build/php]: yeah, I know this is not related to your own change. Your own change seems good regarding existing code (if there is a bug, it was already there before)

**Comment by iSDP** [comment]
Dangit, libsodium 1.0.8 is used, argon2 support was added in 1.0.9, current sodium version is 1.0.16

**Comment by dzuelke** [comment]
The `libsodium23` package is not available in Ubuntu 16.04, which is what the Heroku-16 stack uses as its base image.

**Pull request closed without merging**